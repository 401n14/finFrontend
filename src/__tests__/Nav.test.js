import React from 'react';
import Nav from '../components/NavBar';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { useAuth0 } from '../react-auth0-spa.js';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

configure({adapter:new Adapter()});

const user = {
    email: 'johndoe@me.com',
    email_verified: true,
    sub: 'google-oauth2|2147627834623744883746',
  };

  jest.mock('../react-auth0-spa.js');

describe('Nav Tests', () => {
    beforeEach(() => {
      // Mock the Auth0 hook and make it return a logged in state
      useAuth0.mockReturnValue({
        isAuthenticated: true,
        user,
        logout: jest.fn(),
        loginWithRedirect: jest.fn(),
      });
    });
    it('Renders with required props', async () => {
      const wrapper = mount(
          <BrowserRouter>
      <Nav />
          </BrowserRouter>
      );
      
      expect(wrapper).toBeTruthy();
    });
    it('renders correctly', () => {
        const page = renderer.create(          <BrowserRouter>
            <Nav />
                </BrowserRouter>).toJSON();
        expect(page).toMatchSnapshot();
    });
    it('Logout Button', async () => {
        const wrapper = mount(
            <BrowserRouter>
        <Nav />
            </BrowserRouter>
        );
        let logout = wrapper.find('#qsLogoutBtn').at(5).text();
        expect(logout).toEqual('Log out');
      });
  });