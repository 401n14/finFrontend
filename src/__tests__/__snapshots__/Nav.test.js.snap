// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Tests renders correctly, Snapshot 1`] = `
<div
  className="nav-container"
>
  <nav
    className="navbar navbar-expand-md navbar-light bg-light"
  >
    <div
      className="container"
    >
      <a
        className="logo navbar-brand"
      />
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="mr-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link router-link-exact-active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/chat"
              onClick={[Function]}
            >
              Chat
            </a>
          </li>
        </ul>
        <ul
          className="d-none d-md-block navbar-nav"
        >
          <li
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              href="#"
              id="profileDropDown"
              onClick={[Function]}
            >
              <img
                alt="Profile"
                className="nav-user-profile rounded-circle"
                width="50"
              />
            </a>
            <div
              aria-hidden={true}
              className="dropdown-menu"
              role="menu"
              tabIndex="-1"
            >
              <h6
                className="dropdown-header"
                onClick={[Function]}
                tabIndex="-1"
              />
              <a
                aria-current={null}
                className="dropdown-profile dropdown-item"
                href="/profile"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                 Profile
              </a>
              <button
                className="dropdown-item"
                id="qsLogoutBtn"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                 Log out
              </button>
            </div>
          </li>
        </ul>
        <ul
          className="d-md-none justify-content-between navbar-nav"
          style={
            Object {
              "minHeight": 170,
            }
          }
        >
          <li
            className="nav-item"
          >
            <span
              className="user-info"
            >
              <img
                alt="Profile"
                className="nav-user-profile d-inline-block rounded-circle mr-3"
                width="50"
              />
              <h6
                className="d-inline-block"
              />
            </span>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              href="/profile"
              onClick={[Function]}
            >
              Profile
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="active"
              href="/"
              id="qsLogoutBtn"
              onClick={[Function]}
              style={Object {}}
            >
              Log out
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
